pkg/apis/pipeline/v1beta1/taskrun_defaults.go

```diff
@@ -60,12 +60,66 @@ func (trs *TaskRunSpec) SetDefaults(ctx context.Context) {
 	}
 
 	defaultPodTemplate := cfg.Defaults.DefaultPodTemplate
-	if trs.PodTemplate == nil {
-		trs.PodTemplate = defaultPodTemplate
-	}
+	trs.PodTemplate = mergePodTemplateWithDefault(trs.PodTemplate, defaultPodTemplate)
 
 	// If this taskrun has an embedded task, apply the usual task defaults
 	if trs.TaskSpec != nil {
 		trs.TaskSpec.SetDefaults(ctx)
 	}
 }
+
+func mergePodTemplateWithDefault(tpl, defaultTpl *PodTemplate) *PodTemplate {
+	switch {
+	case defaultTpl == nil:
+		// No configured default, just return the template
+		return tpl
```

Comment:
Could `tpl` be `nil` here? or not possible to have both `defaultTpl` and `tpl` to be `nil` at this point?

Reply:
or don't bother to merge, just return `tpl` here irrespective of it being `nil`

Reply:
Yeah if there is no default, we just return the taskrun template, be it nil or not 😛